cloudbuild/python-scripts/parse_logs.py

```diff
@@ -0,0 +1,513 @@
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+"""Python Script to parse yarn logs to check of job success.
+
+This python file extracts the status of a dataproc job from its job-id and then
+uses this job-id to get the yarn application number.
+The yarn application number enables it to read through the yarn logs to check
+for the number iof records.
+In case the number of records match that of BQ table it returns, in case of
+mismatch, it throws an error.
+"""
+
+from collections.abc import Sequence
+import re
+
+from absl import app
+from google.cloud import bigquery
+from google.cloud import dataproc_v1
+from google.cloud import storage
+
+
+def get_bq_query_result_row_count(client_project_name, query):
+    client = bigquery.Client(project=client_project_name)
+    query_job = client.query(query)
+    query_result = query_job.result()
+    records_read = query_result.total_rows
+    return records_read
+
+
+def get_bq_table_row_count(
+    client_project_name, project_name, dataset_name, table_name, query
+):
+    """Method to get count of rows in a BigQuery Table.
+
+    Args:
+      client_project_name: Name of the project to form the BQ Client
+      project_name: Project ID that contains the table
+      dataset_name: Name of Dataset containing the table
+      table_name: Table Name
+      query: Query string [if any] that was provided to the connector. Incase
+        non-empty, The method executes this query and returns the number of rows
+        obtained.
+
+    Returns:
+      Count of rows in the provided `project_name.dataset_name.table_name` table
+    """
+    if query:
+        return get_bq_query_result_row_count(client_project_name, query)
+    dataset_ref = bigquery.DatasetReference(
+        project=project_name, dataset_id=dataset_name
+    )
+    table_ref = bigquery.TableReference(
+        dataset_ref=dataset_ref, table_id=table_name
+    )
+    client = bigquery.Client(project=client_project_name)
+    table = client.get_table(table=table_ref)
+    row_count = table.num_rows
+    return row_count
+
+
+def extract_metric(logs_as_string, metric_string, end_of_metric_string):
+    """Method to extract the metric value from logs downloaded as a string.
+
+    Args:
+      logs_as_string: Yarn application logs downloaded as a string.
+      metric_string: The string to be found
+      end_of_metric_string: Delimiter at the end of the value.
+
+    Returns:
+      Sum of values of the metric obtained in a file. As there can be 0 or more
+      occurrences of metric in a file
+    """
+    total_metric_sum_in_blob = 0
+    # Keep on finding the metric value as there can be
+    # 1 or more outputs in a log file.
+
+    # The logs are of the format
+    # Number of records read: <value> ;
+    # Here, "Number of records read: " is the metric string
+    # and ";" is the end_of_metric_string.
+    # We find the smallest possible string that matches.
```

Comment:
Why?

Reply:
This is to prevent extra characters from being included in the "value" of the metric. Since the metric is being printed by us, its okay to not add the "?" that limits to the closest match. Either way seems correct to me.

Reply:
Ideally our metrics should only be numbers. So we can have a regex that catches integers or doubles. `.*` is a risky matcher.

Reply:
ACK, Done.